util-linux: musl build fails with "AT_HANDLE_FID undeclared" in sys-utils/nsenter.c

### Describe the bug

After commit 1a1bdcdc2ad22b9f181a9efd05bf451c3936c0cb (`util-linux: update to 2.42`), building `package/utils/util-linux` fails on all musl targets because `sys-utils/nsenter.c` references `AT_HANDLE_FID` without a fallback definition.

`AT_HANDLE_FID` is a Linux 6.5 / glibc 2.39+ symbol. **musl libc does not define it**, even though `linux/nsfs.h` (providing `struct nsfs_file_handle`) is present, which is enough for the meson feature gate `HAVE_STRUCT_NSFS_FILE_HANDLE && HAVE_PIDFD_OPEN` to enable the `USE_NAMESPACE_ID_SUPPORT` code path.

Error:

```
../sys-utils/nsenter.c: In function 'fill_nsfs_file_handle':
../sys-utils/nsenter.c:223:47: error: 'AT_HANDLE_FID' undeclared (first use in this function)
  223 |                               AT_EMPTY_PATH | AT_HANDLE_FID) == -1)
      |                                               ^~~~~~~~~~~~~
```

This is already fixed upstream in util-linux by commit `5452239f6e` (“nsenter: Fix AT_HANDLE_FID on musl”), which simply adds a fallback macro:

```c
#ifndef AT_HANDLE_FID
# define AT_HANDLE_FID 0x200
#endif
```

https://github.com/util-linux/util-linux/commit/5452239f6e69d2d3aaa427d2d2253247cfb7cb7b

Please backport this as [package/utils/util-linux/patches/0002-nsenter-Fix-AT_HANDLE_FID-on-musl.patch](cci:7://file:///D:/App%20Data/Github/immortalwrt/package/utils/util-linux/patches/0002-nsenter-Fix-AT_HANDLE_FID-on-musl.patch:0:0-0:0) until util-linux 2.43 is released.

### OpenWrt version

master, any commit after 0d21956ecdade023981d46eb8f059c3e6a3d00be

### OpenWrt release

SNAPSHOT

### OpenWrt target/subtarget

x86/64

### Device

N/A (build-time failure, toolchain/host-side)

### Image kind

Self-built image

### Steps to reproduce

1. `git clone https://github.com/openwrt/openwrt.git && cd openwrt`
2. `git checkout master` (or any commit ≥ 1a1bdcdc2ad)
3. `./scripts/feeds update -a && ./scripts/feeds install -a`
4. `make defconfig` with `CONFIG_TARGET_x86_64=y` (or any musl target)
5. `make package/utils/util-linux/{clean,compile} V=s`

### Actual behaviour

Build aborts with:

```
../sys-utils/nsenter.c:223:47: error: 'AT_HANDLE_FID' undeclared (first use in this function)
ninja: build stopped: subcommand failed.
make[2]: *** [Makefile:1049: .../util-linux-2.42/.built] Error 1
ERROR: package/utils/util-linux failed to build.
```

### Expected behaviour

util-linux 2.42 compiles cleanly on musl targets.

### Additional info

- Upstream fix (ready to apply as-is): https://github.com/util-linux/util-linux/commit/5452239f6e69d2d3aaa427d2d2253247cfb7cb7b.patch
- Regression introduced in: 1a1bdcdc2ad22b9f181a9efd05bf451c3936c0cb
- Affects: all musl targets (x86_64, aarch64, mipsel, etc.)
- glibc targets (x86_64/glibc, etc.) are not affected since glibc ≥ 2.39 defines `AT_HANDLE_FID`.

### Diffconfig

```text
# default defconfig for x86/64 reproduces; no custom diffconfig required
CONFIG_TARGET_x86=y
CONFIG_TARGET_x86_64=y
CONFIG_TARGET_x86_64_DEVICE_generic=y
```

### Terms

- [x] I am reporting an issue for OpenWrt, not an unsupported fork.

I encountered the same error when compiling the firmware after syncing the code today on the Mediatek platform.


`../sys-utils/nsenter.c: In function 'fill_nsfs_file_handle':
../sys-utils/nsenter.c:223:47: error: 'AT_HANDLE_FID' undeclared (first use in this function)
  223 |                               AT_EMPTY_PATH | AT_HANDLE_FID) == -1)
      |                                               ^~~~~~~~~~~~~
../sys-utils/nsenter.c:223:47: note: each undeclared identifier is reported only once for each function it appears in
ninja: build stopped: subcommand failed.`

--- a/package/utils/util-linux/patches/0001-nsenter-Fix-AT_HANDLE_FID-on-musl.patch
+++ b/package/utils/util-linux/patches/0001-nsenter-Fix-AT_HANDLE_FID-on-musl.patch
@@ -1,0 +1,30 @@
+From 5452239f6e69d2d3aaa427d2d2253247cfb7cb7b Mon Sep 17 00:00:00 2001
+From: Aleksi Hannula <ahannula4+nixgit@gmail.com>
+Date: Tue, 7 Apr 2026 14:52:16 +0300
+Subject: [PATCH] nsenter: Fix AT_HANDLE_FID on musl
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+Signed-off-by: Aleksi Hannula <ahannula4+nixgit@gmail.com>
+---
+ sys-utils/nsenter.c | 3 +++
+ 1 file changed, 3 insertions(+)
+
+diff --git a/sys-utils/nsenter.c b/sys-utils/nsenter.c
+index e10ba9cf90ab..98507958c0f1 100644
+--- a/sys-utils/nsenter.c
++++ b/sys-utils/nsenter.c
+@@ -29,6 +29,9 @@
+ #ifdef HAVE_LINUX_NSFS_H
+ # include <linux/nsfs.h>
+ #endif
++#ifndef AT_HANDLE_FID
++# define AT_HANDLE_FID 0x200
++#endif
+ #ifndef NS_GET_USERNS
+ # define NS_GET_USERNS           _IO(0xb7, 0x1)
+ #endif
+-- 
+2.53.0
+

